code/datums/status_effects/xeno_buffs.dm

```diff
@@ -831,20 +831,17 @@
 
 /datum/status_effect/blessing/warding
 	id = "blessing of warding"
-	///A holder for the exact armor modified by this status effect
 	var/datum/armor/armor_modifier
 
 /datum/status_effect/blessing/warding/on_apply()
-	buff_owner = owner
-	if(!isxeno(buff_owner))
+	if(!isxeno(owner))
 		return FALSE
-	armor_modifier = buff_owner.soft_armor.scaleAllRatings(strength * 2.7)
-	buff_owner.soft_armor = buff_owner.soft_armor.attachArmor(armor_modifier)
+	armor_modifier = getArmor(14 * strength * 2.7 ,3 * strength * 2.7 ,5 * strength * 2.7 ,3 * strength *2.7 ,0 ,0 , 0 ,0 )//create armor for puppets
+	owner.soft_armor=owner.soft_armor.attachArmor(armor_modifier)//we add armor to the puppet
 	return TRUE
 
 /datum/status_effect/blessing/warding/on_remove()
-	buff_owner.soft_armor = buff_owner.soft_armor.detachArmor(armor_modifier)
-	armor_modifier = null
```

Comment:
why'd you remove this? this proc was fine other than referencing owner instead of buff_owner